src/Teambuilder/damagecalc.cpp

```diff
@@ -0,0 +1,868 @@
+    QString oitem = ui->oitem->currentText();
+    QString onature = ui->onature->currentText();
+    QString ostatus = ui->ostatus->currentText();
+    QString ogender = ui->ogender->currentText();
+
+    ui->move->clear();
+    ui->move->addItems(MoveInfo::Names(pgen));
+
+    ui->movetype->clear();
+    ui->movetype->addItems(TypeInfo::Names(pgen));
+
+    ui->movecategory->clear();
+    ui->movecategory->addItems(CategoryInfo::Names());
+
+    ui->weather->clear();
+
+    ui->weather->addItem("Clear Skies", QVariant(Weather::NormalWeather));
+
+    if (gen >= 2) {
+        ui->weather->addItem(whash.value(Weather::Sunny));
+        ui->weather->addItem(whash.value(Weather::Rain));
+        ui->weather->addItem(whash.value(Weather::SandStorm));
+    }
+    if (gen >= 3) {
+        ui->weather->addItem(whash.value(Weather::Hail));
+    }
+    if (gen == 4) {
+        ui->weather->addItem(whash.value(8));
+    }
+    if (gen >= 6) {
+        ui->weather->addItem(whash.value(Weather::StrongSun));
+        ui->weather->addItem(whash.value(Weather::StrongRain));
+        ui->weather->addItem(whash.value(Weather::StrongWinds));
+    }
+
+    ui->mypoke->clear();
+    ui->myability->clear();
+    ui->myitem->clear();
+    ui->mynature->clear();
+    ui->mystatus->clear();
+    ui->mygender->clear();
+
+    ui->opoke->clear();
+    ui->oability->clear();
+    ui->oitem->clear();
+    ui->onature->clear();
+    ui->ostatus->clear();
+    ui->ogender->clear();
+
+    QStringList pokeNames = PokemonInfo::Names(pgen, true);
+
+    ui->mypoke->addItems(pokeNames);
+    ui->opoke->addItems(pokeNames);
+
+    QStringList abilityNames = AbilityInfo::Names(pgen);
+
+    ui->myability->addItems(abilityNames);
+    ui->oability->addItems(abilityNames);
+
+    ui->myitem->addItems(ItemInfo::SortedNames(pgen));
+    ui->oitem->addItems(ItemInfo::SortedNames(pgen));
+
+    ui->mynature->addItems(NatureInfo::Names(pgen));
+    ui->onature->addItems(NatureInfo::Names(pgen));
+
+    ui->mystatus->addItems(StatInfo::StatusNames());
+    ui->ostatus->addItems(StatInfo::StatusNames());
+
+    int max = (gen >= 3 ? 31 : 15);
+
+    ui->myhpiv->setMaximum(max);
+    ui->myatkiv->setMaximum(max);
+    ui->mydefiv->setMaximum(max);
+    ui->myspatkiv->setMaximum(max);
+    ui->myspdefiv->setMaximum(max);
+    ui->myspdiv->setMaximum(max);
+
+    ui->ohpiv->setMaximum(max);
+    ui->oatkiv->setMaximum(max);
+    ui->odefiv->setMaximum(max);
+    ui->ospatkiv->setMaximum(max);
+    ui->ospdefiv->setMaximum(max);
+    ui->ospdiv->setMaximum(max);
+
+    ui->myhpiv->setValue(max);
+    ui->myatkiv->setValue(max);
+    ui->mydefiv->setValue(max);
+    ui->myspatkiv->setValue(max);
+    ui->myspdefiv->setValue(max);
+    ui->myspdiv->setValue(max);
+
+    ui->ohpiv->setValue(max);
+    ui->oatkiv->setValue(max);
+    ui->odefiv->setValue(max);
+    ui->ospatkiv->setValue(max);
+    ui->ospdefiv->setValue(max);
+    ui->ospdiv->setValue(max);
+
+    ui->move->setCurrentText(move);
```

Comment:
Is it possible to use setEditText() insted of setCurrentText() for these lines (200 - 216)? Because Travis is compiling with Qt 4.8(?) and setCurrentText was implemented in Qt 5.